### DISCUSSION TITLE
How to model Supertype/Subtype for Junction Relationships?

### DISCUSSION THREAD

**Discussion by arkmech** [prisma/prisma#15513] [Q&A] [answered]
Objective:
I want to model Supertype/Subtype, but doesn't seem possible with Prisma.
Supertype/Subtytpe relations are 1:1.

`Party Model`
```prisma
model Party {
  id                  String                @id @default(uuid())
  PartyClassification PartyClassification[]
}
```

`Party Type Model`
```prisma
model PartyType {
  id                  String                @id @default(uuid())
  name                String
  PartyClassification PartyClassification[]
}
```

`Party Classification Model` (Junction Table between Party and Party Type)
```prisma
model PartyClassification {
  Party                               Party                       @relation(fields: [partyId], references: [id])
  partyId                            String                      
  PartyType                       PartyType               @relation(fields: [partyType], references: [id])
  partyType                       String                      
  OrganizationClassificationPartyId   OrganizationClassification? @relation("partyId")
  OrganizationClassificationPartyType OrganizationClassification? @relation("partyType")

  @@id([partyId, partyType])
}
```

`Organization Classification` (Subtype of Junction Table PartyClassification)
```prisma
model OrganizationClassification {
  PartyClassificationPartyId   PartyClassification @relation("partyId", fields: [partyId], references: [partyId])
  partyId                      String
  PartyClassificationPartyType PartyClassification @relation("partyType", fields: [partyType], references: [partyType])
  partyType                    String
  organizationClassificationSpecificField String

  @@id([partyId, partyType])
}
```

**Issue:**
```
The argument `references` must refer to a unique criteria in the related model. Consider adding an `@unique` attribute to the field `partyId` in the model `PartyClassification`.
A one-to-one relation must use unique fields on the defining side. Either add an `@unique` attribute to the field `partyType`
```

Pretty much need to add `@unique` constraint on both `partyId` and `partyType` on `PartyClassification` AND `partyId` and `partyType` on `OrganizationClassification` to resolve issue.

But that isn't right as in `PartyClassification`, `partyId` and `partyType` are many to many and not unique same with `OrganizationClassification`

Shouldn't the `@@id([partyId, partyType])`, also imply that the keys are only unique together, thus this issue should not arise?

**Comment by arkmech** [comment, marked as answer]
Resolved:
```prisma
model OrganizationClassification {
  PartyClassification PartyClassification @relation(fields: [partyId, partyType], references: [partyId, partyType])
  partyId             String
  partyType           String

  @@id([partyId, partyType])
}
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
